flow/workflows/qrep_flow.go

```diff
@@ -254,6 +279,33 @@ func (q *QRepFlowExecution) waitForNewRows(ctx workflow.Context, lastPartition *
 	return nil
 }
 
+func (q *QRepFlowExecution) signalHandler(v shared.CDCFlowSignal) {
```

Comment:
is there some way to share this shutdown/pause logic?

Also seems like `activeSignal` shouldn't be switched from `ShutdownSignal` to `PauseSignal`. So should be checking activeSignal to prevent that sort of state transition

Reply:
The data structures associated with `CDCFlow` and `QRepFlow` are different enough for sharing the logic to be not trivial.

I don't think there's a way to transition from `ShutdownSignal` to `PauseSignal`, as we only process one signal at a time and the return from shutdown happens immediately.

Reply:
Could have utility function in shared that takes signal value + active signal value & returns new active signal value